Added REMOVE_SUBMISSIONS flag

Nedlinin opened the pull request:
... to allow us to toggle between reporting a post and removing it.  This currently is set on the config level globally but at some point we probably want to determine this dynamically based on the type of error (for instance, a feature is in "beta").

Changed verb from "archive" to "remove" in function names to more closely reflect the nomenclature of reddit.

Fixes #7.

Nedlinin commented:
You'll have to take it easy on all my updates.  I tried to do this with no Python environment setup on this machine so I couldn't actually run any of this.  Using the automatic checking you've got setup to lint my code at the moment 🥇

Nedlinin commented on main.py:
```diff
@@ -39,65 +39,64 @@ def _run(self):
             if self.database.known_submission(submission) is True:
                 continue
+            else:
+                self.database.save_submission(submission)
```
Not sure we actually want this here specifically.  I noticed we had a few edge cases where we were not saving the submission so I moved this up to here but now I realize that if a service fails we will have saved that submission and skip it the next time.

I'll have to look into how PRAW works further.  Is it even possible it gives us the same submission more than once?

martijnboers replied:
By default it will get the last 100 submissions and stream new posts

martijnboers replied:
Looking at it now it might the edge case might have been this:

https://github.com/martijnboers/BlotterTrax/pull/13/files#diff-5bc02cefb3ea9e27f1a6776eabd1935dR70

I prefer not saving here and explicitly calling the save method following all possible paths because of the reason you mention

Nedlinin replied:
Yep.  I have since merged that into this branch so the above code snippet is outdated/unused.

Nedlinin commented on main.py:
```diff
@@ -39,65 +39,64 @@ def _run(self):
+                    self._perform_exceeds_threshold_mod_action(submission, last_fm_service)
+                    continue
             except pylast.WSError:
-                continue
```
We previously failed out if LastFM couldn't connect.  Instead, I think we continue on and just know we may have missed a check somewhere.

Nedlinin replied:
Definitely possible (and arguably likely for sure).  But in that case we simply won't get the nice LastFM lookup reply post but will have still done anything we could at the time.

Nedlinin commented:
Think this is in a better place now.  Only saves posts when an action was taken (be it removing, commenting, etc.).

Nedlinin commented:
I should note before this is merged that I have done a grand total of *zero* testing this.  Might be good to spin it up on your test sub before merging :)

martijnboers commented:
@Nedlinin Could you have a look at the merge conflicts? I will do another functional test after

Nedlinin commented:
@martijnboers : Dealt with conflicts and fixed another small issue found (trying to access count instead of listener_count for a ServiceResult).

Lastly I removed some URL parsing from the main function as the work I did to fix the Youtube parsing on another PR basically duplicated that work anyway.

Take another look and let me know what you think.

martijnboers commented:
LGTM
